learningequality/studio
# Sync endpoint robustifying and node creation bug fix

## Description

* Adds tests for Channel viewset
* Adds tests for ContentNode viewset
* Refactors sync endpoint to delegate change handling to viewsets
* Makes CRUD and sync handling parallel in viewsets
* Create mixins for opt-in bulk processing and change event handling
* Handles ContentNode creation in a way that does not mess up the orphanage tree
* Adds read/write permissions to orphanage tree for all authenticated users
* Adds regression test for bug for creation and copying events

## Steps to Test

- Create a topic or resource on a fresh DB
- Make sure the sync endpoint returns a 200
